health.go

```diff
@@ -48,34 +48,49 @@ type health struct {
 	checksListener CheckListeners
 	healthListener HealthListeners
 	lock           sync.RWMutex
+
+	// Check config defaults
+	defaultExecutionPeriod  time.Duration
+	defaultInitialDelay     time.Duration
+	defaultInitiallyPassing bool
 }
 
-func (h *health) RegisterCheck(cfg *Config) error {
-	if cfg.Check == nil || cfg.Check.Name() == "" {
-		return errors.Errorf("misconfigured check %v", cfg.Check)
+func (h *health) RegisterCheck(check Check, opts ...CheckOption) error {
+	if check == nil || check.Name() == "" {
+		return errors.Errorf("misconfigured check %v", check)
+	}
+
+	cfg := checkConfig{
```

Comment:
please extract the config initialization, including the options apply

Reply:
Extract where, why?

Reply:
Extract a function that creates and initializes the config struct

Reply:
I made the requested changes.